src/cmdhandler/command_handler.cpp

```diff
@@ -210,29 +217,32 @@ std::string command_handler::state_system(const std::vector<std::string>& tokens
 
 std::string command_handler::collector_system(const std::vector<std::string>& tokens, StateVariables& current_state) {
     if(tokens[0] == "list"){
-        std::string response = "Current collectors:";
+        std::stringstream response;
+        response << "Current collectors:";
 
         for(auto const& collector : current_state.collectors){
-            response += "\n    "+collector.first+": "+collector.second;
+            response << "\n    " + collector.first << ": " << collector.second;
         }
 
-        return response;
+        return response.str();
     }else if(tokens[0] == "set"){
-        if(tokens.size() != 4){
-            return "please provide a collector name and an ip\n    ex: set collector gcs 127.0.0.1";
+        if(tokens.size() != 5){
+            return "please provide a collector name, ip, and port\n    ex: set collector gcs 127.0.0.1 53";
         }
 
         std::string collector_id = tokens[2];
 
         asio::error_code ec;
-        asio::ip::address ip = asio::ip::make_address(tokens[3], ec);
+        asio::ip::address address = asio::ip::make_address(tokens[3], ec);
 
         if(ec){
             return "please provide a valid ipv4 address";
         }
 
-        current_state.collectors.insert(std::pair<std::string, std::string>(collector_id, ip.to_string()));
-        return collector_id+" added with ip "+ip.to_string();
+        asio::ip::udp::endpoint endpoint(address, std::stoi(tokens[4]));
```

Comment:
If I pass "f" (or any non int) as the port for a collector it stops the server.
Maybe add a check & alert user if a non valid port has been given.